packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -977,8 +977,8 @@ export class IndexedDbPersistence implements Persistence {
         // to make sure it gets a chance to run.
         this.markClientZombied();
 
-        if (isSafari() && navigator.appVersion.match(/Version\/1[45]/)) {
-          // On Safari 14 and 15, we do not run any cleanup actions as it might
+        if (isSafari() && (navigator.appVersion.match(/(?:Version|Mobile)\/1[456]/) || navigator.userAgent.match(/(?:Version|Mobile)\/1[456]/))) {
```

Comment:
Please run `yarn prettier` in the `packages/firestore` directory to format the code. It will do something wonky to try to get line wrapping at 80 characters.